Compile script improvement

S0nFra opened the pull request:
In this branch I have implemented an improved version of the script for compiling the application without a gui.

Specifically, I added a --help and the ability to be able to choose which architecture to compile for.

Initially I also planned to add the ability to compile in DEBUG mode, which was later removed with the last commit. We can consider whether to give this possibility or not.

S0nFra committed:
enhance compilation script for MQTT Shell with architecture options and help information

S0nFra committed:
refactor build script by removing redundant build flags

freedreamer82 commented:
seems a good point but i would like to have also other version with no gui , gui and mixed.
maybe we can create a script in the same folder of  cmd/mqtt-gui cmd/mqtt-no-gui cmd/mqttt-gui where you can selcet the version u want to compile 

example.

./compile.sh --mode gui  .--debug ..

freedreamer82 commented:
i would create a general script compiling no-gui and gui ..is possibile u can extend the functionality of this script including the other cmd/versions ?

S0nFra committed:
Add build script for easy cross-compilation

Introduced a new build script to simplify compiling the application for multiple architectures and operating systems. This enhancement streamlines the build process, making it easier for developers to generate binaries for different platforms with minimal effort.

S0nFra committed:
Add support for extra parameters in compile script

S0nFra commented:
I introduced a build script to simplify the project compilation process.

Main changes:

- Added support for building on multiple architectures and operating systems (cross-compilation)
- Enabled passing extra parameters to the compile script for greater flexibility
- Improved documentation and help messages for easier usage
- Removed redundant build flags from the compilation process
- These updates make it easier to generate binaries for different platforms and improve the maintainability of the build process.

Let me know if you have any suggestions or feedback!
